src/bin/hal/cmd/message.rs

```diff
@@ -116,7 +116,7 @@ fn exec_verify<'a>(matches: &clap::ArgMatches<'a>) {
 	}
 	let recid = secp256k1::recovery::RecoveryId::from_i32((sig_bytes[0] - 27 & 0x03) as i32)
 		.expect("invalid recoverable signature (invalid recid)");
-	let compressed = sig_bytes[0] & 0x04 != 0x04;
+	let compressed = (sig_bytes[0] - 27 & 0x04) != 0;
```

Comment:
At least to my eye, parentheses would make this clearer. I suppose it matches what's above, though.